src/Codeception/Lib/InnerBrowser.php

```diff
@@ -1063,20 +1063,39 @@ protected function getFormFor(Crawler $node)
      */
     protected function getFormValuesFor(Form $form)
     {
+        $formNodeCrawler = new Crawler($form->getFormNode());
         $values = [];
         $fields = $form->all();
         foreach ($fields as $field) {
-            if ($field instanceof FileFormField || $field->isDisabled() || !$field->hasValue()) {
+            if ($field instanceof FileFormField || $field->isDisabled()) {
                 continue;
             }
+
+            if (!$field->hasValue()) {
+                // if unchecked a checkboc and if there is hidden input with same name to submit unchecked value
+                $hiddenInput = $formNodeCrawler->filter('input[type=hidden][name="'.$field->getName().'"]:not([disabled])');
+                if (!count($hiddenInput)) {
+                    continue;
+                } else {
+                    // there might be multiple hidden input with same name, but we will only grab last one's value
+                    $hiddenFieldValue = $hiddenInput->last()->attr('value');
+                }
+            }
+
```

Comment:
Almost there, fields with `[]` don't benefit from your change yet.

I think that it would be the best to get fieldValue at this point.
```php
if (!$field->hasValue()) {
   ...
        $fieldValue = $hiddenInput->last()->attr('value');
    }
}  else {
   $fieldValue = $field->getValue();
```

and replace `$field->getValue();` with `$fieldValue` in all places below.

`$values[$fieldName][] = $field->getValue();` to `$values[$fieldName][] = $fieldValue;`

and your new block would become simply `$values[$fieldName] = $fieldValue;`

Reply:
Fixed in fd66e5d2f18d6b2c3facc93023e56c3b88a1b47d